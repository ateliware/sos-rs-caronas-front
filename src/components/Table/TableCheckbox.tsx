type Props = {
  header?: boolean;
  checked?: boolean;
  onChange?: (checked: boolean) => void;
  onAction?: () => void;
};

function ActionButton(props: Pick<Props, 'onAction'>) {
  if (!props.onAction) return null;
  return (
    <button
      onClick={props.onAction}
      className="btn btn--icon-square btn--table-action ml-s-50"
    >
      <div className="btn__body">
        <span className="material-symbols-outlined btn__body__icon">
          arrow_drop_down
        </span>
      </div>
    </button>
  );
}

export default function TableCheckbox(props: Props) {
  if (props.header) {
    return (
      <th className="table__cell table__cell--checkbox">
        <div className="table__cell__content">
          <div className="form-check">
            <input
              checked={props.checked}
              onChange={(event) => props.onChange?.(event.target.checked)}
              className="form-check-input"
              type="checkbox"
              id="checkbox"
            />
          </div>
          <ActionButton onAction={props.onAction} />
        </div>
      </th>
    );
  } else {
    return (
      <td className="table__cell table__cell--checkbox">
        <div className="table__cell__content">
          <div className="form-check">
            <input
              checked={props.checked}
              onChange={(event) => props.onChange?.(event.target.checked)}
              className="form-check-input"
              type="checkbox"
              id="checkbox"
            />
          </div>
        </div>
      </td>
    );
  }
}
