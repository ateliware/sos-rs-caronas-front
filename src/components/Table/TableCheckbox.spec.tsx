import { fireEvent, render } from '@testing-library/react';

import TableCheckbox from './TableCheckbox';

describe('TableCheckbox', () => {
  it('should render', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <TableCheckbox />
          </tr>
        </tbody>
      </table>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variants', () => {
    const action = jest.fn();
    const change = jest.fn();
    const { container } = render(
      <table>
        <thead>
          <tr>
            <TableCheckbox checked header onAction={action} onChange={change} />
          </tr>
        </thead>
      </table>
    );
    expect(container).toMatchSnapshot();

    const checkbox = container.querySelector('.form-check-input') as Element;
    fireEvent.click(checkbox);
    expect(change).toHaveBeenCalledWith(false);

    const actionBtn = container.querySelector('.btn--table-action') as Element;
    fireEvent.click(actionBtn);
    expect(action).toHaveBeenCalled();
  });
});
