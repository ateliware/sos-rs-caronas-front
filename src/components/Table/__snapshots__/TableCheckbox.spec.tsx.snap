// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCheckbox should render 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="table__cell table__cell--checkbox"
        >
          <div
            class="table__cell__content"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="checkbox"
                type="checkbox"
              />
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableCheckbox should render with variants 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th
          class="table__cell table__cell--checkbox"
        >
          <div
            class="table__cell__content"
          >
            <div
              class="form-check"
            >
              <input
                checked=""
                class="form-check-input"
                id="checkbox"
                type="checkbox"
              />
            </div>
            <button
              class="btn btn--icon-square btn--table-action ml-s-50"
            >
              <div
                class="btn__body"
              >
                <span
                  class="material-symbols-outlined btn__body__icon"
                >
                  arrow_drop_down
                </span>
              </div>
            </button>
          </div>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
